browser/brave_ads/ads_tab_helper.cc

```diff
@@ -150,6 +166,11 @@ void AdsTabHelper::DidFinishLoad(content::RenderFrameHost* render_frame_host,
   TabUpdated();
 }
 
+void AdsTabHelper::OnRetrieveSearchResultAd(SearchResultAdsList search_ads) {
+  // TODO(https://github.com/brave/brave-browser/issues/20852):
```

Comment:
why is the tab helper responsible for this?

Reply:
The idea was to call AdsService::OnRetrieveSearchResultAds here. But you are right, we could do this call at SearchResultAdHandler class